compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -768,34 +768,36 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let mut err = self.demand_suptype_diag(expr.span, expected_ty, actual_ty).unwrap();
             let lhs_ty = self.check_expr(&lhs);
             let rhs_ty = self.check_expr(&rhs);
-            if self.can_coerce(lhs_ty, rhs_ty) {
-                if !lhs.is_syntactic_place_expr() {
-                    // Do not suggest `if let x = y` as `==` is way more likely to be the intention.
-                    if let hir::Node::Expr(hir::Expr {
-                        kind: ExprKind::Match(_, _, hir::MatchSource::IfDesugar { .. }),
-                        ..
-                    }) = self.tcx.hir().get(
-                        self.tcx.hir().get_parent_node(self.tcx.hir().get_parent_node(expr.hir_id)),
-                    ) {
-                        // Likely `if let` intended.
-                        err.span_suggestion_verbose(
-                            expr.span.shrink_to_lo(),
-                            "you might have meant to use pattern matching",
-                            "let ".to_string(),
-                            Applicability::MaybeIncorrect,
-                        );
-                    }
+            let (mut msg, applicability, eq) = if self.can_coerce(rhs_ty, lhs_ty) {
+                ("likely", Applicability::MachineApplicable, true)
+            } else {
+                ("might have", Applicability::MaybeIncorrect, false)
```

Comment:
I wonder if `Applicability::MaybeIncorrect` should have a `Display` or some sort to convert to text. Still `likely` feels weird, I see another alternative is `probably` which may be a bit better (my two cents).

Reply:
Addressed, now we only use "might have" regardless of applicability.